### PR TITLE
Support https requests without certificate authentication

### PR DESCRIPTION
**Pull request by yeajung**
Sample command for endpoint with HTTPS protocol:
`./perf-top-linux --dashboard NodeAnalysis --endpoint https://localhost:9600`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lib/perf-top/util/generate-data.js [hunk 1]
```diff
@@ -73,8 +76,13 @@ function makeHttpRequest (httpOptions, done) {
       done(rawData);
     });
   };
-
-  var req = http.request(httpOptions, respond);
+  var protocol = http;
```
Comment (ditac) [inline lib/perf-top/util/generate-data.js]: I think client might be a better name for this variable.

File: lib/perf-top/util/generate-data.js [hunk 2]
```diff
@@ -73,8 +76,13 @@ function makeHttpRequest (httpOptions, done) {
...
+  if (urlOptions.protocol == "https:") {
+    protocol = https;
+    urlOptions.rejectUnauthorized = false;
+  }
+  var protocol = (urlOptions.protocol == "https:") ? https : http;
```
Comment (ditac) [inline lib/perf-top/util/generate-data.js]: We seem to be doing this twice. We can remove the if block

File: lib/perf-top/util/generate-data.js (hunk 1 above)
Comment (yeajung) [reply lib/perf-top/util/generate-data.js]: Done

File: lib/perf-top/util/generate-data.js (hunk 2 above)
Comment (yeajung) [reply lib/perf-top/util/generate-data.js]: Oops. Removed the repetition.

**Pull request merged**